multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -1252,30 +1252,32 @@ def _top_k_average(
                     #  "Model soups: averaging weights of multiple fine-tuned models improves accuracy without
                     #  increasing inference time", https://arxiv.org/pdf/2203.05482.pdf
                     monitor_op = {MIN: operator.le, MAX: operator.ge}[minmax_mode]
-
-                    logger.info(f"Start to fuse {len(top_k_model_paths)} checkpoints via the greedy soup algorithm.")
-
                     ingredients = [top_k_model_paths[0]]
-                    self._model = self._load_state_dict(
-                        model=model,
-                        path=top_k_model_paths[0],
-                        prefix=prefix,
-                    )
-                    best_score = self.evaluate(val_df, [validation_metric_name])[validation_metric_name]
-                    for i in range(1, len(top_k_model_paths)):
-                        cand_avg_state_dict = average_checkpoints(
-                            checkpoint_paths=ingredients + [top_k_model_paths[i]],
+                    if len(ingredients) > 1:
```

Comment:
Should be `if len(top_k_model_paths) > 1:` ?

Reply:
Good point. It was a bug.